### PR TITLE
DB Enhancement

### PR DESCRIPTION
**Pull request by Arrow768**
Adds foreign keys to various tables.
Adds the game_id to the ban, notes and warnings table.
Adds the char_id to the ss13_death table.
Changes ss13_death to use NULL instead of empty strings for missing values.
Changes the ss13_death table to use ckeys instead of keys

### CODE REVIEW

File: code/controllers/subsystems/statistics.dm [hunk 1]
```diff
@@ -233,68 +233,42 @@
...
+		if(!query.Execute(list(
+			"name"=H.real_name,
+			"ckey"=H.ckey,
+			"char_id"=H.character_id,
+			"job"=H.mind:assigned_role,
+			"special"=H.mind:special_role,
+			"pod"=podname,
+			"tod"=time2text(world.realtime, "YYYY-MM-DD hh:mm:ss"),
+			"laname"=H.lastattacker:real_name,
+			"lackey"=H.lastattacker:ckey,
```
Comment (skull132) [inline code/controllers/subsystems/statistics.dm:264]:
```suggestion
			"laname"=H.lastattacker.real_name,
			"lackey"=H.lastattacker.ckey,
```

Comment (skull132) [reply code/controllers/subsystems/statistics.dm:264]: Hopefully this will work?

File: code/controllers/subsystems/statistics.dm [hunk 2]
```diff
@@ -233,68 +233,42 @@
...
+			"job"=H?.mind.assigned_role,
+			"special"=H?.mind.special_role,
+			"pod"=podname,
+			"tod"=time2text(world.realtime, "YYYY-MM-DD hh:mm:ss"),
+			"laname"=H?.lastattacker.real_name,
+			"lackey"=H?.lastattacker.ckey,
+			"gender"=H.gender,
+			"bruteloss"=H.getBruteLoss(),
+			"fireloss"=H.getFireLoss(),
+			"brainloss"=H.getBrainLoss(),
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/statistics.dm:268]: This is kind of pointless now? When people die they always die with 200 brainloss

File: code/modules/admin/admin_attack_log.dm [hunk 3]
```diff
@@ -1,5 +1,5 @@
-/mob/var/lastattacker = null
-/mob/var/lastattacked = null
+/mob/var/mob/lastattacker = null
+/mob/var/mob/lastattacked = null
```
Comment (NonQueueingMatt) [inline code/modules/admin/admin_attack_log.dm:2]: is this handled in destroy at all?

File: code/controllers/subsystems/statistics.dm (hunk 2 above)
Comment (Arrow768) [reply code/controllers/subsystems/statistics.dm:268]: Not necessarily as this applies to all species as well as borgs.

File: code/modules/admin/admin_attack_log.dm (hunk 3 above)
Comment (Arrow768) [reply code/modules/admin/admin_attack_log.dm:2]: I didn't check, as this is somewhat out of the scope for that pr.

**Pull request merged**